flutter_local_notifications/lib/src/flutter_local_notifications_plugin.dart

```diff
@@ -118,6 +119,22 @@ class FlutterLocalNotificationsPlugin {
     if (kIsWeb) {
       return true;
     }
+    if (Platform.isAndroid && initializationSettings.android == null) {
```

Comment:
The plugin already does platform checks down below so wouldn't it better to shift the null checks to happen within those if statements? This avoids checking the platform twice. Note that tests are also currently failing too as calls like `Platform.isAndroid` depends on the environment used to run the code. This is why the preferred way to do platform checks is via using `defaultTargetPlatform` as this can be overridden during tests. In case it helps, this is done by other plugins too (e.g. https://github.com/firebase/flutterfire/search?q=defaultTargetPlatform). By shifting the null checks to occur within the existing if statements, this should avoid the duplication I mentioned and fix the tests